docs/ff-concepts/adding-customization/common-examples.md

```diff
@@ -0,0 +1,284 @@
+
+#### Example 1: Creating a new Instance of `ProductStruct`
+To create a new `ProductStruct` instance, initialize it with the required properties:
+
+```js
+// Create a new instance of ProductStruct
+final newProduct = ProductStruct(
+    productId: '123',
+    name: 'Example Product',
+    description: 'A sample product description.',
+    category: 'Electronics',
+    subCategory: 'Mobile Phones',
+    price: PriceStruct(amount: 299.99, currency: 'USD'),
+    sizes: ['Small', 'Medium', 'Large'],
+    colors: [ColorsStruct(colorName: 'Red', colorHex: '#FF0000')],
+    images: [ImagesStruct(thumbnail: 'https://example.com/image.jpg')],
+    stockStatus: StockStatusStruct(xs: 0, small: 2),
+    reviews: [ReviewsStruct(rating: 4, comment: 'Great product!')],
+);
+
+```
+
+#### Example 2: Accessing Properties of an Existing `ProductStruct` object
+
+If you have an existing `ProductStruct` object (e.g., retrieved from a list of products), you can access its properties or return specific values back to the calling Action. Let's assume you have an Action that calls a Custom Action to retrieve a field value from the provided `ProductStruct` object.
+
+- **Returning a Single Field from ProductStruct**
+
+This function retrieves and returns the product's name. The return type is `String?` to account for the possibility of a null value.
+```js
+// Function to return the product name from a ProductStruct instance
+String? getProductName(ProductStruct product) {
+    // Access and return the product name
+    return product.name;
+}
+```
+- **Returning a List of Review Comments from ProductStruct**
+
+This function retrieves a list of review comments from the reviews field in the `ProductStruct`. The return type is `List<String>` as it returns a list of comments (or an empty list if there are no reviews).
+
+```js
+// Function to return a list of review comments from a ProductStruct instance
+List<String> getProductReviewComments(ProductStruct product) {
+  // Check if reviews are present and return a list of review comments
+  return product.reviews?.map((review) => review.comment ?? '').toList() ?? [];
+}
+```
+
+#### Example 3: Modifying Properties of an Existing `ProductStruct` Object
+You can also modify the properties of an existing `ProductStruct` object. This can be helpful if you want to update a field before saving the data back to Firebase or passing it into an action.
+
+- **Simple Property Modification**
+In this example, we’ll modify a single property, like `productId`, of an existing `ProductStruct` object. This example is straightforward and demonstrates how to update a basic field in the object.
+
+```js
+// Function to update the product ID of a ProductStruct instance
+Future updateProductId(ProductStruct product, String newProductId) {
```

Comment:
Instead of showing an example of updating the product ID, let's use another property like product name or description, as a product's ID is typically intended to remain unchanged in most scenarios.

Reply:
Great catch! Updated